src/components/common/SearchInput/SearchInput.tsx

```diff
@@ -40,26 +43,45 @@ const SearchInput = forwardRef<SearchInputRef, SearchInputProps>(
     { onSearchClick, buttonTestId = 'store-search-button', ...props },
     ref
   ) {
+    const [searchQuery, setSearchQuery] = useState<string>('')
+    const [suggestionsOpen, setSuggestionsOpen] = useState<boolean>(false)
```

Comment:
WDYT about using the new react [useDefferedValue](https://reactjs.org/docs/hooks-reference.html#usedeferredvalue)?